rust/src/metrics/reader.rs

```diff
@@ -0,0 +1,78 @@
+use crate::wit::wasi::otel::metrics::collect;
+use opentelemetry_sdk::{
+    error::{OTelSdkError, OTelSdkResult},
+    metrics::{
+        data::ResourceMetrics, reader::MetricReader, InstrumentKind, ManualReader, Temporality,
+    },
+};
+use std::sync::Arc;
+
+#[derive(Debug)]
+pub struct WasiMetricReader {
+    reader: Arc<ManualReader>,
```

Comment:
It wasn't obvious to me why we needed to embed the manual reader initially. A comment saying something like we embed the manual reader to avoid reimplementing too much OpenTelemetry SDK internals.